transactions/transactions_service.go

```diff
@@ -321,14 +321,28 @@ func (svc *transactionsService) SendKeysend(ctx context.Context, amount uint64,
 
 	// the payment definitely succeeded
 	now := time.Now()
-	dbErr := svc.db.Model(&dbTransaction).Updates(map[string]interface{}{
-		"State":          constants.TRANSACTION_STATE_SETTLED,
-		"PaymentHash":    paymentHash,
-		"Preimage":       &preimage,
-		"FeeMsat":        &fee,
-		"FeeReserveMsat": 0,
-		"SettledAt":      &now,
-	}).Error
+
+	dbErr := svc.db.Transaction(func(tx *gorm.DB) error {
+		// Because we don't know the payment hash in advance when creating the original keysend transaction,
+		// there may have been another transaction created with the same payment hash by the nwc_payment_sent event.
+		// Delete that one and update the one we already created.
+		// This would not be needed if we always can pass the payment hash when doing keysend.
+		err := tx.Delete(&db.Transaction{}, "payment_hash = ?", paymentHash).Error
+		if err != nil {
+			return err
+		}
```

Comment:
hmm, is there no other way. feels like a hack and just thrown code at it.

Reply:
I think in LNDHub we create the preimage and the payment hash

Reply:
@bumi we can do this if all the connectors that support passing a payment hash to the keysend method.

- More LDK custom changes on our fork to pass a payment hash rather then letting LDK generate one
- Disable Keysend on Breez and Greenlight since they don't conform and we don't want to spend time on them

Then Keysend will only be supported by LDK and LND.

What do you think?